app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js

```diff
@@ -2,75 +2,92 @@ import { permissionRpcMethods } from '@metamask/permission-controller';
 import { selectHooks } from '@metamask/snaps-rpc-methods';
 import { hasProperty } from '@metamask/utils';
 import { ethErrors } from 'eth-rpc-errors';
-import { UNSUPPORTED_RPC_METHODS } from '../../../../shared/constants/network';
-import localHandlers from './handlers';
+import { handlers as localHandlers, legacyHandlers } from './handlers';
 
 const allHandlers = [...localHandlers, ...permissionRpcMethods.handlers];
 
-const handlerMap = allHandlers.reduce((map, handler) => {
-  for (const methodName of handler.methodNames) {
-    map[methodName] = handler;
-  }
-  return map;
-}, {});
+// The primary home of RPC method implementations in MetaMask. MUST succeed
```

Comment:
[nit] succeed may be a confusing word here
```suggestion
// The primary home of RPC method implementations in MetaMask. MUST follow
```